### PR TITLE
fix ChatMessageChunk concat error

### PR DESCRIPTION
**Pull request by anexplore**
- Description: fix ChatMessageChunk '+' error, add __add__  method for ChatMessageChunk
  - Issue: ChatMessageChunk has required field "role", however it's use BaseMessageChunk.__add__method which do not handle role field
  ```python
      # BaseMessageChunk
      def __add__(self, other: Any) -> BaseMessageChunk:  # type: ignore
        if isinstance(other, BaseMessageChunk):
            # If both are (subclasses of) BaseMessageChunk,
            # concat into a single BaseMessageChunk
            
           # this line will raise error for role not set
            return self.__class__(
                content=self.content + other.content,
                additional_kwargs=self._merge_kwargs_dict(
                    self.additional_kwargs, other.additional_kwargs
                ),
            )
        else:
            raise TypeError(
                'unsupported operand type(s) for +: "'
                f"{self.__class__.__name__}"
                f'" and "{other.__class__.__name__}"'
            )
  ````

### CODE REVIEW

**Review by eyurtsev — COMMENTED** [review]
@anexplore Thanks for the fix, would you be able to add a unit test to verify that it works correctly?

**Comment by baskaryan** [comment]
i think this issue exists with all child classes of BaseMessageChunk, cc @nfcampos

File: libs/langchain/langchain/schema/messages.py [hunk 1]
```diff
@@ -221,7 +221,21 @@ def type(self) -> str:
 class ChatMessageChunk(ChatMessage, BaseMessageChunk):
     """A Chat Message chunk."""
 
-    pass
+    def __add__(self, other: Any) -> BaseMessageChunk:
+        if isinstance(other, BaseMessageChunk):
+            return self.__class__(
+                content=self.content + other.content,
+                role=self.role,
```
Comment (nfcampos) [inline libs/langchain/langchain/schema/messages.py:228]: What if they have two different roles?

**Comment by nfcampos** [comment]
@baskaryan how do you mean? the other subclasses don't have additional args?

**Comment by baskaryan** [comment]
> @baskaryan how do you mean? the other subclasses don't have additional args?

```python
class HumanMessage(BaseMessage):
    """A Message from a human."""

    example: bool = False
    """Whether this Message is being passed in to the model as part of an example 
        conversation.
    """

    @property
    def type(self) -> str:
        """Type of the message, used for serialization."""
        return "human"


class HumanMessageChunk(HumanMessage, BaseMessageChunk):
    """A Human Message chunk."""

    pass
```

some messages have `example`, FunctionMessage has `name`. for human/ai msg probs doesn't matter bc generated stuff won't be examples (and won't be human msg) but Functions name could matter? and more generally if we add any other params to subclasses in future it's not obv with this method being inherited that something would need to be updated

**Pull request closed without merging**